File: Algorithm/Code_Materials/Java/src/main/java/com/yugutou/charpter19_dp/level2/FindLengthOfLCIS.java
package com.yugutou.charpter19_dp.level2;

/**
 * LeetCode674.给定一个未经排序的整数数组，
 * 找到最长且连续递增的子序列，并返回该序列的长度。
 */
public class FindLengthOfLCIS {
    public static void main(String[] args) {
        int[] nums = {1, 3, 5, 4, 7};
        System.out.println(findLengthOfLCIS(nums));
    }

    public static int findLengthOfLCIS(int[] nums) {
        int[] dp = new int[nums.length];
        for (int i = 0; i < dp.length; i++) {
            dp[i] = 1;
        }
        int res = 1;
        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i + 1] > nums[i]) {
                dp[i + 1] = dp[i] + 1;
            }
            res = res > dp[i + 1] ? res : dp[i + 1];
        }
        return res;

    }
}

File: Algorithm/Code_Materials/c_or_cpp/charpter16_slide/level1/findLength_of_lCIS.cpp
#include <stdio.h>

int findLengthOfLCIS(int* nums, int numsSize) {
    int left = 0, right = 0, res = 0;
    while (right < numsSize) {
        // 右侧的新元素比左侧的小，则重新开始记录left位置
        if (right > 0 && nums[right - 1] >= nums[right]) {
            left = right;
        }
        right++;
        res = res > right - left ? res : right - left;
    }
    return res;
}

int main() {
    int nums[] = {1, 3, 5, 7, 9};
    int numsSize = sizeof(nums) / sizeof(nums[0]);
    int result = findLengthOfLCIS(nums, numsSize);
    printf("Length of Largest Clique: %d\n", result);
    return 0;
}